src/main/kotlin/apply/application/EvaluationService.kt

```diff
@@ -0,0 +1,86 @@
+package apply.application
+
+import apply.domain.evaluation.Evaluation
+import apply.domain.evaluation.EvaluationRepository
+import apply.domain.evaluation.dto.EvaluationResponse
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import javax.annotation.PostConstruct
+import javax.transaction.Transactional
+
+@Transactional
+@Service
+class EvaluationService(
+    private val evaluationRepository: EvaluationRepository,
+    private val recruitmentService: RecruitmentService
+) {
+    private fun findAll(): List<Evaluation> {
+        return evaluationRepository.findAll()
+    }
+
+    private fun findById(id: Long): Evaluation? {
+        if (id == 0L) return null
+
+        return evaluationRepository.findByIdOrNull(id) ?: throw IllegalArgumentException("해당 id의 평가를 찾을 수 없습니다.")
+    }
+
+    fun findAllWithRecruitment(): List<EvaluationResponse> {
+        return findAll().map {
+            EvaluationResponse(
+                it.id,
+                it.title,
+                it.description,
+                recruitmentService.getById(it.recruitment).title,
+                it.recruitment,
+                findById(it.beforeEvaluation)?.title ?: "이전 평가 없음",
+                it.beforeEvaluation
+            )
+        }.toList()
+    }
+
+    fun deleteById(id: Long) {
+        evaluationRepository.deleteById(id)
+
+        resetBeforeEvaluationContain(id)
+    }
+
+    private fun resetBeforeEvaluationContain(id: Long) {
+        evaluationRepository.findAll()
+            .firstOrNull { it.hasSameBeforeEvaluationWith(id) }?.resetBeforeEvaluation()
+    }
+
+    @PostConstruct
+    private fun populateDummy() {
+        if (evaluationRepository.count() != 0L) {
+            return
+        }
+        val evaluations = listOf(
+            Evaluation(
+                "프리코스 대상자 선발",
+                "[리뷰 절차]\n" +
+                    "https://github.com/woowacourse/woowacourse-docs/tree/master/precourse",
+                1L
+            ),
+            Evaluation(
+                " 1주차 - 숫자야구게임",
+                "[리뷰 절차]\n" +
+                    "https://github.com/woowacourse/woowacourse-docs/tree/master/precourse",
+                1L
+            ),
```

Comment:
`secondEvaluation!!.beforeEvaluation = 1L`은 아래와 같이 줄일 수 있어요.

```suggestion
            Evaluation(
                "1주차 - 숫자야구게임",
                "[리뷰 절차]\n" +
                    "https://github.com/woowacourse/woowacourse-docs/tree/master/precourse",
                recruitment = 1L,
                beforeEvaluation = 1L
            ),
```

Reply:
반영했습니다. 
set으로 넣어주기 보다는 생성자로 넣는게 더 올바르고 이해하기 쉬운 방식이었네요😂